lib/gis_robot_suite/structural_metadata_builder.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module GisRobotSuite
+  class StructuralMetadataBuilder
+    def self.build(*args)
+      new(*args).build
+    end
+
+    # @param [String] druid
+    # @param [Hash<Symbol,Assembly::ObjectFile>] objects
+    def initialize(cocina_model, druid, objects)
+      @cocina_model = cocina_model
+      @druid = druid
+      @objects = objects
+    end
+
+    attr_reader :cocina_model, :druid, :objects
+
+    def build
+      cocina_model.structural.new(contains: file_sets)
+    end
+
+    private
+
+    def file_access
+      @file_access ||= cocina_model.access.to_h
+                                   .slice(:view, :download, :location, :controlledDigitalLending)
+                                   .tap do |access|
+        access[:view] = 'dark' if access[:view] == 'citation-only'
+      end
+    end
+
+    # @return [Array<Cocina::Models::FileSet>]
+    def file_sets
+      objects.compact_blank.map.with_index(1) do |(key, value), seq|
```

Comment:
Can you give "key" and "value" more descriptive names?

Reply:
done in third commit